### ISSUE TITLE
overriden_super_call triggers on super called within guard's else statement

### ISSUE THREAD

**Issue by krze** [realm/SwiftLint#2160]
Hello, I'm upgrading to 0.25.1 from quite an old version. After working through fixing some valid violations I found an issue while running the rule `overriden_super_call`. In `Rules.md`, this is listed as a violation:

```swift
class VC: UIViewController {
	override func viewWillAppear(_ animated: Bool) {↓
		super.viewWillAppear(animated)
		//Other code
		super.viewWillAppear(animated)
	}
}

```

Which makes perfect sense. However, I have a method which calls super in a guard's else statement. Super is only called once, whether the guard passes or fails.

In effect, the code looks something like this (working off the violation example):

```swift
class VC: UIViewController {
	override func viewWillAppear(_ animated: Bool) {↓
                guard foo = self.foo else {
	             super.viewWillAppear(animated)
	             return
                }

		//Other code
		super.viewWillAppear(animated)
	}
}
```

**Comment by SDGGiesbrecht** [comment]
@krze,

Logic branches (`if`, `guard`, `switch`, etc.) can become arbitrarily complex. I doubt they can be supported very easily. I do not know whether it is better to ignore methods containing branching logic, or enforce the rule naïvely in spite of the branching logic (which is the behaviour you see right now).

But two things might help:

1. Can you refactor to call only once by using `defer`?
```swift
override func viewWillAppear(_ animated: Bool) {

    defer { super.viewWillAppear(animated) }

    guard foo = self.foo else {
        return
    }

    // Other code
}
```

2. If not, you can always use `// swiftlint:disable`.
```swift
override func viewWillAppear(_ animated: Bool) { // swiftlint:disable:this overriden_super_call
    guard foo = self.foo else {
        super.viewWillAppear(animated)
        return
    }

    //Other code
    super.viewWillAppear(animated)
}
```

**Comment by krze** [comment]
Defer did the trick, thanks for the help!
